provisioning/provisioning.go

```diff
@@ -60,6 +67,14 @@ func (cfg *SubmitConfig) SetDeleteSession(session bool) {
 	cfg.DeleteSession = session
 }
 
+// SetAuth sets the IAuthenticator that will be used
+func (cfg *SubmitConfig) SetAuth(a auth.IAuthenticator) {
```

Comment:
Here we should Validate Authenticator before setting it?

Reply:
Validate what? An authenticator cannot be in an invalid state. It can be misconfigured, however this is not something that makes sense to check on setting it here, because it only makes sense in the context of the wider universe and would in effect be validated when the api requiest is actually sent. We would not gain anything by checking for it here (just add aditional delays for the extra requersts to test the settings).
  And any problems with input configuration are caught by `Configure()`.